app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -466,7 +466,12 @@ public function isBoltOnlyPayment()
      */
     private function isCustomerGroupDisabled($customerGroupId)
     {
-        $disabledCustomerGroups = explode(',', Mage::getStoreConfig('payment/boltpay/bolt_disabled_customer_groups'));
+        $disabledCustomerGroups = Mage::getStoreConfig('payment/boltpay/bolt_disabled_customer_groups');
+        // Case ot empty groups
```

Comment:
// Case **of** empty groups

Also, it is better to use this comment to explain why you must return false here, which is really because you are preventing the case of the user not belonging to a customer group and no customer groups being disabled.

Possibly, easier to understand code would be:
```
     // False if the customer is not part of any groups or no groups are disabled
     if (empty($customerGroupId) || empty($disabledCustomerGroups) {
          return false;
     }
     $disabledCustomerGroups = explode(',', $disabledCustomerGroups);
     return in_array($customerGroupId, $disabledCustomerGroups);
```

Reply:
@mccottry PHP has magic!
explode(',', '') returns array('') and this array is not empty!
in_array(0, array('')) returns true
Type will be fixed.

Reply:
correct.  In the code listed above, as you have written the method, the `empty($disabledCustomerGroups)` is not testing an array, but rather a string.  If you review the code above, you will see that you will never be allowed the opportunity to compare `in_array(0, array(''))`.

Thank you for updating the spelling.